src/components/CommentModal.tsx

```diff
@@ -76,58 +79,63 @@ export const CommentModal: React.FC<CommentModalProps> = ({
 
   return (
     <>
-      <input type="checkbox" id={`modal-${id}`} className="modal-toggle" />
-      <label htmlFor={`modal-${id}`} className="modal z-10 cursor-pointer">
-        <label
-          className="card-body modal-box relative m-[-10px] w-11/12 max-w-5xl rounded-md"
-          htmlFor=""
+      {postx && (
+        <ReactModal
+          closeTimeoutMS={100}
+          isOpen={isOpen}
+          overlayClassName="fixed inset-0 z-20 bg-black/30"
+          className="absolute top-1/2 left-1/2 z-40 max-h-[95vh] w-11/12 max-w-5xl -translate-x-1/2 -translate-y-1/2 overflow-y-auto overflow-x-hidden rounded-md border-slate-500 bg-white p-5 outline-none md:w-1/2"
+          shouldCloseOnEsc
+          shouldCloseOnOverlayClick
+          onRequestClose={() => setIsOpen(false)}
         >
```

Comment:
Refactored to use `react-modal` instead of the DaisyUI modal because it was too difficult to control the the open/close state of the daisy modal from the search bar. React-modal lets us use component state so it's much easier to manage